# Quality Control

  Note that Monocle assume the gene expresison is log-normally distributed. So here we perform some filtering and then give it a check. ^[Monocle is tuned for Single-Cell RNA-seq, so does the quality control described in the [Recommended Analysis Protocol](http://cole-trapnell-lab.github.io/monocle-release/docs/#recommended-analysis-protocol)]. And it won't intervere following analysis since in this bulk analysis, we are not facing issues of Single-Cell protocol like *low mRNA cell sample* or *doublets or triplets*, etc. 

## gene filtering{#gf}
```{r include=FALSE}
options(useFancyQuotes = FALSE)
dt <- fread("C:/Users/Ryan/Documents/3.Kidney/HUMAN_all.gene_RPKM.xls")
dt <- dt[!duplicated(dt$external_gene_name)]

setcolorder(dt,
            c(names(dt)[1:7],
              names(dt)[-(1:7)][order(names(dt)[-(1:7)])])
)

expr <- as.matrix(dt[,-(1:7)])
expr <- expr[,grep("pre|post", colnames(expr))]
gene_ann <- data.frame(dt[,1:7])
setnames(gene_ann, "external_gene_name", "gene_short_name")
rownames(gene_ann) = dt$ensembl_gene_id
rownames(expr) <- dt$ensembl_gene_id
sample_ann <- data.frame(sampleID = colnames(expr), person = substr(colnames(expr), 15,16),
                         time = substr(colnames(expr), 18, nchar(colnames(expr))))
rownames(sample_ann) = sample_ann$sampleID

fd <- new("AnnotatedDataFrame", data = gene_ann)
pd <- new("AnnotatedDataFrame", data = sample_ann)


human <- newCellDataSet(expr, phenoData = pd, featureData = fd,
                        expressionFamily = tobit())

```
```{r}
human <- detectGenes(human, min_expr = 0.1)
plot(density(fData(human)$num_cells_expressed))
```
The number of cells express a certain gene is bimodal distributed, for $threshold = .1$, some genes are expressed in all 163 samples, and some genes are only expressed in a little few samples.
```{r}
expressed.gene <- row.names(subset(fData(human), num_cells_expressed >= 50))
```
* Genes has >= 50 samples with $RPKM >= .1$ are kept. (22047/54514 genes left)

***
## check log-nomality
  As suggest in Monocle mannual, take log(rpkm + 1), scale each log-transformed gene expression(subtract the mean and divided by $sd$), make the whole log-rpkm matrix a single vector then compare it to a standard normal.
```{r}
L <- log(exprs(human[expressed.gene,]) + 1)
```

Apparently, our filtered data pass the log-normality test.
```{r, echo=F}
melt.L.scaled <- melt(t(scale(t(L))))
ggplot(melt.L.scaled, aes(x=value)) + geom_density(aes(color = "scaled log-expr")) +
  stat_function(aes(color = "std. normal"), fun = dnorm, size = 0.5) +
  xlab('Standardized log(Expression)') + ylab('Density') +
  scale_colour_manual("Legend", values = c("black", "red")) +
  ggtitle("Scaled log-expression v.s. Std.normal")
```


```{r, echo = F}
melt.L <- melt(L)
ggplot(melt.L, aes(x=value)) + geom_density(aes(color = "log-expr")) +
  stat_function(aes(color = "std. normal"), fun = dnorm, size = 0.5) +
  xlab('Standardized log(Expression)') + ylab('Density') +
  scale_colour_manual("Legend", values = c("black", "red")) +
  ggtitle("Non-scale log-expression v.s. Std.normal")

```
