<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A Report Example</title>
  <meta name="description" content="My first attemp to write scientific report via bookdown.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="A Report Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My first attemp to write scientific report via bookdown." />
  <meta name="github-repo" content="Yaphets-Bo/kidneyproject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Report Example" />
  
  <meta name="twitter:description" content="My first attemp to write scientific report via bookdown." />
  

<meta name="author" content="Bo Sun">


<meta name="date" content="2018-02-09">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="homologs-reduction-of-de-gene-set.html">
<link rel="next" href="gene-expression-patterns-on-kidney-injury-across-species.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequsites</a></li>
<li class="chapter" data-level="2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html"><i class="fa fa-check"></i><b>2</b> t-SNE analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-human"><i class="fa fa-check"></i><b>2.1</b> Dataset Human</a><ul>
<li class="chapter" data-level="2.1.1" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#post-samples-form-a-distinguishable-cluster"><i class="fa fa-check"></i><b>2.1.1</b> POST samples form a distinguishable cluster</a></li>
<li class="chapter" data-level="2.1.2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#independent-t-sne-on-each-time-point"><i class="fa fa-check"></i><b>2.1.2</b> Independent t-SNE on each time point</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-mouse"><i class="fa fa-check"></i><b>2.2</b> Dataset Mouse</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a><ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#gf"><i class="fa fa-check"></i><b>3.1</b> gene filtering</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#check-log-nomality"><i class="fa fa-check"></i><b>3.2</b> check log-nomality</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html"><i class="fa fa-check"></i><b>4</b> Pseudo-time trajectory prediction</a><ul>
<li class="chapter" data-level="4.1" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-1-2-set-ordering-genes-and-reduce-demisionality"><i class="fa fa-check"></i><b>4.1</b> Step 1-2: Set ordering genes and reduce demisionality</a></li>
<li class="chapter" data-level="4.2" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-3-ordering-the-cells-in-pseudotime"><i class="fa fa-check"></i><b>4.2</b> Step 3: ordering the cells in pseudotime</a></li>
<li class="chapter" data-level="4.3" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#two-dimensional-case"><i class="fa fa-check"></i><b>4.3</b> Two-dimensional case</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html"><i class="fa fa-check"></i><b>5</b> Homologs Reduction of DE gene set</a><ul>
<li class="chapter" data-level="5.1" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#download-orthology-information-from-biomart"><i class="fa fa-check"></i><b>5.1</b> Download orthology information from BioMart</a></li>
<li class="chapter" data-level="5.2" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#focus-de-genes-of-human-and-mouse-on-homologous-genes-only"><i class="fa fa-check"></i><b>5.2</b> Focus DE genes of Human and Mouse on homologous genes only</a><ul>
<li class="chapter" data-level="5.2.1" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#take-iri2h-iri4h-as-early-response"><i class="fa fa-check"></i><b>5.2.1</b> Take IRI2h + IRI4H as “early response”</a></li>
<li class="chapter" data-level="5.2.2" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#take-iri2h-as-early-reponse-only"><i class="fa fa-check"></i><b>5.2.2</b> Take IRI2h as early reponse only</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><i class="fa fa-check"></i><b>6</b> Heatmap Viewing of Informative Genes Used to Predict Sample Trajectory (#hm)</a><ul>
<li class="chapter" data-level="6.1" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html#transform-rpkm-to-rpc"><i class="fa fa-check"></i><b>6.1</b> Transform RPKM to RPC</a></li>
<li class="chapter" data-level="6.2" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html#set-ordering-genes-and-reduce-demisionality"><i class="fa fa-check"></i><b>6.2</b> Set ordering genes and reduce demisionality</a></li>
<li class="chapter" data-level="6.3" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html#ordering-the-samples-in-pseudotimetime"><i class="fa fa-check"></i><b>6.3</b> Ordering the samples in pseudotime/time</a></li>
<li class="chapter" data-level="6.4" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm.html#heatmap-samples-ordered-by-pseudotime"><i class="fa fa-check"></i><b>6.4</b> heatmap, samples ordered by Pseudotime</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="gene-expression-patterns-on-kidney-injury-across-species.html"><a href="gene-expression-patterns-on-kidney-injury-across-species.html"><i class="fa fa-check"></i><b>7</b> Gene expression patterns on kidney injury across species</a><ul>
<li class="chapter" data-level="7.1" data-path="gene-expression-patterns-on-kidney-injury-across-species.html"><a href="gene-expression-patterns-on-kidney-injury-across-species.html#species-dominate-the-clustering"><i class="fa fa-check"></i><b>7.1</b> Species Dominate the clustering</a></li>
<li class="chapter" data-level="7.2" data-path="gene-expression-patterns-on-kidney-injury-across-species.html"><a href="gene-expression-patterns-on-kidney-injury-across-species.html#variance-decomposition"><i class="fa fa-check"></i><b>7.2</b> Variance Decomposition</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Report Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory-hm" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Heatmap Viewing of Informative Genes Used to Predict Sample Trajectory (#hm)</h1>
<div id="transform-rpkm-to-rpc" class="section level2">
<h2><span class="header-section-number">6.1</span> Transform RPKM to RPC</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">human &lt;-<span class="st"> </span><span class="kw">newCellDataSet</span>(<span class="kw">as.matrix</span>(expr),
                        <span class="dt">phenoData =</span> pd,
                        <span class="dt">featureData =</span> fd,
                        <span class="dt">lowerDetectionLimit =</span> <span class="fl">0.1</span>,
                        <span class="dt">expressionFamily =</span> <span class="kw">tobit</span>(<span class="dt">Lower =</span> <span class="fl">0.1</span>))

rpc_matrix &lt;-<span class="st"> </span><span class="kw">relative2abs</span>(human, <span class="dt">method =</span> <span class="st">&quot;num_genes&quot;</span>)

human &lt;-<span class="st"> </span><span class="kw">newCellDataSet</span>(<span class="kw">as</span>(<span class="kw">as.matrix</span>(rpc_matrix), <span class="st">&quot;sparseMatrix&quot;</span>),
                        <span class="dt">phenoData =</span> pd,
                        <span class="dt">featureData =</span> fd,
                        <span class="dt">lowerDetectionLimit =</span> <span class="fl">0.5</span>,
                        <span class="dt">expressionFamily =</span> <span class="kw">negbinomial.size</span>())
human &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(human)
human &lt;-<span class="st"> </span><span class="kw">estimateDispersions</span>(human)</code></pre></div>
<pre><code>## Warning: Deprecated, use tibble::rownames_to_column() instead.</code></pre>
<pre><code>## Removing 77 outliers</code></pre>
<p>expr(RPKMS), phenoData(samples) and featureData(genes) are pre-processed in the back end, from the original RPKM file. And then transform RPKM data to mRNA counts data(RPC = RNA per cell), which make the use of negative binomial distribution possible and thence claimed more suitable for this Monocle 2 software.</p>
<div class="rmdcaution">
<p>
The transformation was done by an algorithm called <em>Census</em>. And you can find the whole documentation here at <a href="http://cole-trapnell-lab.github.io/monocle-release/docs/#converting-tpm-fpkm-values-into-mrna-counts-alternative">Monocle 2</a>.
</p>
</div>
</div>
<div id="set-ordering-genes-and-reduce-demisionality" class="section level2">
<h2><span class="header-section-number">6.2</span> Set ordering genes and reduce demisionality</h2>
<blockquote>
<p>&quot; Monocle looks for genes that vary in “interesting” (i.e. not just noisy) ways, and uses these to structure the data.&quot;</p>
<p>— Monocle Mannual</p>
</blockquote>
<p>This is the <em>feature selection</em> step in the Monocle’s machine learning algorithm for this job, which has major impact of the shape of the trajectory to be constructed<span class="citation">(Trapnell <a href="#ref-R-monocle">2017</a>)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">human &lt;-<span class="st"> </span><span class="kw">detectGenes</span>(human, <span class="dt">min_expr =</span> <span class="fl">0.1</span>)
expressed.gene &lt;-<span class="st"> </span><span class="kw">row.names</span>(<span class="kw">subset</span>(<span class="kw">fData</span>(human), num_cells_expressed <span class="op">&gt;=</span><span class="st"> </span><span class="dv">35</span>))
diff_test_res &lt;-<span class="st"> </span><span class="kw">differentialGeneTest</span>(human[expressed.gene,],
                                      <span class="dt">fullModelFormulaStr =</span> <span class="st">&quot;~time + person&quot;</span>, <span class="dt">cores =</span> <span class="dv">2</span>)
ordering_genes &lt;-<span class="st"> </span><span class="kw">row.names</span> (<span class="kw">subset</span>(diff_test_res, qval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.000001</span>))
human &lt;-<span class="st"> </span><span class="kw">setOrderingFilter</span>(human, ordering_genes)
<span class="kw">plot_ordering_genes</span>(human)</code></pre></div>
<p><img src="bookdown_files/figure-html/rpc_de-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">human &lt;-<span class="st"> </span><span class="kw">reduceDimension</span>(human, <span class="dt">max_components =</span> <span class="dv">2</span>,
                         <span class="dt">method =</span> <span class="st">&#39;tSNE&#39;</span>)
human &lt;-<span class="st"> </span><span class="kw">orderCells</span>(human,<span class="dt">reverse =</span> T)</code></pre></div>
<p>The gene selection algorithm a machine learning algorithm inside monocle 2, and the model ~time + person means the gene expression was modeled as a linear function of time point and person(individual difference), then due to the variability of the human dataset, tens of genes tend to be varied on different time point and individual.</p>
<p>Therefore I tried to decrease the number of genes entering in the heatmap by very stringent q-value cut-off(&lt;1e-6).</p>
<p>157 of 12518 de genes pass the q-value cut-off and are used to plot the heat map. (from a total of 22047 genes that are defined express “interestingly” as described in Section <a href="quality-control.html#gf">3.1</a>).</p>
</div>
<div id="ordering-the-samples-in-pseudotimetime" class="section level2">
<h2><span class="header-section-number">6.3</span> Ordering the samples in pseudotime/time</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_cell_trajectory</span>(human, <span class="dt">color_by =</span> <span class="st">&quot;time&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:rpcTime"></span>
<img src="bookdown_files/figure-html/rpcTime-1.png" alt="Sample trajectory colored by real time points" width="672" />
<p class="caption">
FIGURE 6.1: Sample trajectory colored by real time points
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_cell_trajectory</span>(human, <span class="dt">color_by =</span> <span class="st">&quot;Pseudotime&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:rpcPseudotime"></span>
<img src="bookdown_files/figure-html/rpcPseudotime-1.png" alt="Same trajectory colored by Pseudotime" width="672" />
<p class="caption">
FIGURE 6.2: Same trajectory colored by Pseudotime
</p>
</div>
</div>
<div id="heatmap-samples-ordered-by-pseudotime" class="section level2">
<h2><span class="header-section-number">6.4</span> heatmap, samples ordered by Pseudotime</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ordering.expr &lt;-<span class="st"> </span>rpc_matrix[ordering_genes,<span class="kw">grep</span>(<span class="st">&quot;post&quot;</span>,<span class="kw">rownames</span>(<span class="kw">pData</span>(human)))][,<span class="kw">order</span>(<span class="kw">pData</span>(human)[<span class="kw">grep</span>(<span class="st">&quot;post&quot;</span>,<span class="kw">rownames</span>(<span class="kw">pData</span>(human))),]<span class="op">$</span>Pseu)]
ordering.expr &lt;-<span class="st"> </span><span class="kw">log</span>(ordering.expr <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)


<span class="kw">pheatmap</span>(
  <span class="dt">mat               =</span> ordering.expr,
  <span class="dt">scale             =</span> <span class="st">&quot;row&quot;</span>,
  <span class="dt">color             =</span> <span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set3&quot;</span>),
  <span class="dt">border_color      =</span> <span class="ot">NA</span>,
  <span class="dt">show_colnames     =</span> <span class="ot">FALSE</span>,
  <span class="dt">show_rownames     =</span> T,
  <span class="co">#annotation_col    = col_ann,</span>
  <span class="co">#annotation_colors = color_ann,</span>
  <span class="dt">cluster_cols      =</span> <span class="ot">FALSE</span>,
  <span class="dt">drop_levels       =</span> <span class="ot">TRUE</span>,
  <span class="dt">fontsize          =</span> <span class="dv">14</span>,
  <span class="dt">main              =</span> <span class="st">&quot;Scaled log-rpc heatmap</span><span class="ch">\n</span><span class="st">POST sample ordered by Pseudotime&quot;</span>
)</code></pre></div>
<div class="figure">
<img src="bookdown_files/figure-html/good_hm-1.png" alt="columns ordered by rank(Pseudotime)" width="672" />
<p class="caption">
(#fig:good_hm)columns ordered by rank(Pseudotime)
</p>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-R-monocle">
<p>Trapnell, Cole. 2017. <em>Monocle: Clustering, Differential Expression, and Trajectory Analysis for Single- Cell Rna-Seq</em>. <a href="https://bioconductor.org/packages/release/bioc/html/monocle.html" class="uri">https://bioconductor.org/packages/release/bioc/html/monocle.html</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="homologs-reduction-of-de-gene-set.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-expression-patterns-on-kidney-injury-across-species.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
