<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A Report Example</title>
  <meta name="description" content="My first attemp to write scientific report via bookdown.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="A Report Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My first attemp to write scientific report via bookdown." />
  <meta name="github-repo" content="Yaphets-Bo/kidneyproject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Report Example" />
  
  <meta name="twitter:description" content="My first attemp to write scientific report via bookdown." />
  

<meta name="author" content="Bo Sun">


<meta name="date" content="2017-12-06">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="quality-control.html">
<link rel="next" href="homologs-reduction-of-de-gene-set.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequsites</a></li>
<li class="chapter" data-level="2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html"><i class="fa fa-check"></i><b>2</b> t-SNE analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-human"><i class="fa fa-check"></i><b>2.1</b> Dataset Human</a></li>
<li class="chapter" data-level="2.2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-mouse"><i class="fa fa-check"></i><b>2.2</b> Dataset Mouse</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a><ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#gf"><i class="fa fa-check"></i><b>3.1</b> gene filtering</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#check-log-nomality"><i class="fa fa-check"></i><b>3.2</b> check log-nomality</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html"><i class="fa fa-check"></i><b>4</b> Pseudo-time trajectory prediction</a><ul>
<li class="chapter" data-level="4.1" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-1-2-set-ordering-genes-and-reduce-demisionality"><i class="fa fa-check"></i><b>4.1</b> Step 1-2: Set ordering genes and reduce demisionality</a></li>
<li class="chapter" data-level="4.2" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-3-ordering-the-cells-in-pseudotime"><i class="fa fa-check"></i><b>4.2</b> Step 3: ordering the cells in pseudotime</a></li>
<li class="chapter" data-level="4.3" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#two-dimensional-case"><i class="fa fa-check"></i><b>4.3</b> Two-dimensional case</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html"><i class="fa fa-check"></i><b>5</b> Homologs Reduction of DE gene set</a><ul>
<li class="chapter" data-level="5.1" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#download-orthology-information-from-biomart"><i class="fa fa-check"></i><b>5.1</b> Download orthology information from BioMart</a></li>
<li class="chapter" data-level="5.2" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#focus-de-genes-of-human-and-mouse-on-homologous-genes-only"><i class="fa fa-check"></i><b>5.2</b> Focus DE genes of Human and Mouse on homologous genes only</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Report Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pseudo-time-trajectory-prediction" class="section level1">
<h1><span class="header-section-number">4</span> Pseudo-time trajectory prediction</h1>
<div id="step-1-2-set-ordering-genes-and-reduce-demisionality" class="section level2">
<h2><span class="header-section-number">4.1</span> Step 1-2: Set ordering genes and reduce demisionality</h2>
<blockquote>
<p>&quot; Monocle looks for genes that vary in “interesting” (i.e. not just noisy) ways, and uses these to structure the data.&quot;</p>
<p>— Monocle Mannual</p>
</blockquote>
<p>This is the <em>feature selection</em> step in the Monocle’s machine learning algorithm for this job, which has major impact of the shape of the trajectory to be constructed<span class="citation">(Trapnell <a href="#ref-R-monocle">2017</a>)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">diff_test_res &lt;-<span class="st"> </span><span class="kw">differentialGeneTest</span>(human[expressed.gene,],
                                      <span class="dt">fullModelFormulaStr =</span> <span class="st">&quot;~time&quot;</span>)

ordering_genes &lt;-<span class="st"> </span><span class="kw">row.names</span> (<span class="kw">subset</span>(diff_test_res, qval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.01</span>))
ordering_genes &lt;-<span class="st"> </span><span class="kw">intersect</span>(ordering_genes, expressed.gene)
human &lt;-<span class="st"> </span><span class="kw">setOrderingFilter</span>(human, ordering_genes)
human &lt;-<span class="st"> </span><span class="kw">reduceDimension</span>(human, <span class="dt">max_components =</span> <span class="dv">2</span>,
                            <span class="dt">method =</span> <span class="st">&#39;DDRTree&#39;</span>)
human &lt;-<span class="st"> </span><span class="kw">orderCells</span>(human)</code></pre></div>
<p>17985 of 22047 genes left. (The 22047 genes are the genes that express “interestingly” as described in Section <a href="quality-control.html#gf">3.1</a>).</p>
</div>
<div id="step-3-ordering-the-cells-in-pseudotime" class="section level2">
<h2><span class="header-section-number">4.2</span> Step 3: ordering the cells in pseudotime</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_cell_trajectory</span>(human, <span class="dt">color_by =</span> <span class="st">&quot;time&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:time"></span>
<img src="_main_files/figure-html/time-1.png" alt="Sample trajectory colored by real time points" width="672" />
<p class="caption">
Figure 4.1: Sample trajectory colored by real time points
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_cell_trajectory</span>(human, <span class="dt">color_by =</span> <span class="st">&quot;Pseudotime&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:pseudotime"></span>
<img src="_main_files/figure-html/pseudotime-1.png" alt="Same trajectory colored by Pseudotime" width="672" />
<p class="caption">
Figure 4.2: Same trajectory colored by Pseudotime
</p>
</div>
<p>Monocle doesn’t know a priori which of the trajectory of the tree to call the “beginning”. So the Pseudotime is plotted reversly. It’s a little bit tricky to make it right. But in our case, it’s clear that the left is beginning and the right is end since the left points are pre and right points are post, as suggested in the previous figuer.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(
  <span class="kw">pData</span>(human)[,<span class="kw">c</span>(<span class="st">&quot;person&quot;</span>, <span class="st">&quot;time&quot;</span>, <span class="st">&quot;Pseudotime&quot;</span>)], <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;A table of the Pseudotime generated&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-9">Table 4.1: </span>A table of the Pseudotime generated</caption>
<thead>
<tr class="header">
<th></th>
<th align="left">person</th>
<th align="left">time</th>
<th align="right">Pseudotime</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>sample.kidney_01_baseline_post</td>
<td align="left">01</td>
<td align="left">baseline_post</td>
<td align="right">16.343957</td>
</tr>
<tr class="even">
<td>sample.kidney_01_baseline_pre</td>
<td align="left">01</td>
<td align="left">baseline_pre</td>
<td align="right">18.759666</td>
</tr>
<tr class="odd">
<td>sample.kidney_02_baseline_post</td>
<td align="left">02</td>
<td align="left">baseline_post</td>
<td align="right">15.905122</td>
</tr>
<tr class="even">
<td>sample.kidney_02_baseline_pre</td>
<td align="left">02</td>
<td align="left">baseline_pre</td>
<td align="right">23.558735</td>
</tr>
<tr class="odd">
<td>sample.kidney_03_baseline_post</td>
<td align="left">03</td>
<td align="left">baseline_post</td>
<td align="right">15.546759</td>
</tr>
<tr class="even">
<td>sample.kidney_03_baseline_pre</td>
<td align="left">03</td>
<td align="left">baseline_pre</td>
<td align="right">23.977949</td>
</tr>
<tr class="odd">
<td>sample.kidney_04_baseline_post</td>
<td align="left">04</td>
<td align="left">baseline_post</td>
<td align="right">3.471537</td>
</tr>
<tr class="even">
<td>sample.kidney_04_baseline_pre</td>
<td align="left">04</td>
<td align="left">baseline_pre</td>
<td align="right">37.794212</td>
</tr>
<tr class="odd">
<td>sample.kidney_05_baseline_post</td>
<td align="left">05</td>
<td align="left">baseline_post</td>
<td align="right">19.294851</td>
</tr>
<tr class="even">
<td>sample.kidney_05_baseline_pre</td>
<td align="left">05</td>
<td align="left">baseline_pre</td>
<td align="right">37.268242</td>
</tr>
<tr class="odd">
<td>sample.kidney_06_baseline_post</td>
<td align="left">06</td>
<td align="left">baseline_post</td>
<td align="right">19.417621</td>
</tr>
<tr class="even">
<td>sample.kidney_06_baseline_pre</td>
<td align="left">06</td>
<td align="left">baseline_pre</td>
<td align="right">36.971777</td>
</tr>
<tr class="odd">
<td>sample.kidney_07_baseline_post</td>
<td align="left">07</td>
<td align="left">baseline_post</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td>sample.kidney_07_baseline_pre</td>
<td align="left">07</td>
<td align="left">baseline_pre</td>
<td align="right">36.049794</td>
</tr>
<tr class="odd">
<td>sample.kidney_08_baseline_post</td>
<td align="left">08</td>
<td align="left">baseline_post</td>
<td align="right">19.006155</td>
</tr>
<tr class="even">
<td>sample.kidney_08_baseline_pre</td>
<td align="left">08</td>
<td align="left">baseline_pre</td>
<td align="right">20.959575</td>
</tr>
<tr class="odd">
<td>sample.kidney_09_baseline_post</td>
<td align="left">09</td>
<td align="left">baseline_post</td>
<td align="right">2.493296</td>
</tr>
<tr class="even">
<td>sample.kidney_09_baseline_pre</td>
<td align="left">09</td>
<td align="left">baseline_pre</td>
<td align="right">36.291851</td>
</tr>
<tr class="odd">
<td>sample.kidney_10_baseline_post</td>
<td align="left">10</td>
<td align="left">baseline_post</td>
<td align="right">3.972422</td>
</tr>
<tr class="even">
<td>sample.kidney_10_baseline_pre</td>
<td align="left">10</td>
<td align="left">baseline_pre</td>
<td align="right">37.515398</td>
</tr>
<tr class="odd">
<td>sample.kidney_11_baseline_post</td>
<td align="left">11</td>
<td align="left">baseline_post</td>
<td align="right">14.915194</td>
</tr>
<tr class="even">
<td>sample.kidney_11_baseline_pre</td>
<td align="left">11</td>
<td align="left">baseline_pre</td>
<td align="right">26.426196</td>
</tr>
<tr class="odd">
<td>sample.kidney_12_baseline_post</td>
<td align="left">12</td>
<td align="left">baseline_post</td>
<td align="right">15.076965</td>
</tr>
<tr class="even">
<td>sample.kidney_12_baseline_pre</td>
<td align="left">12</td>
<td align="left">baseline_pre</td>
<td align="right">37.998821</td>
</tr>
<tr class="odd">
<td>sample.kidney_13_baseline_post</td>
<td align="left">13</td>
<td align="left">baseline_post</td>
<td align="right">4.481699</td>
</tr>
<tr class="even">
<td>sample.kidney_13_baseline_pre</td>
<td align="left">13</td>
<td align="left">baseline_pre</td>
<td align="right">26.022365</td>
</tr>
<tr class="odd">
<td>sample.kidney_14_baseline_post</td>
<td align="left">14</td>
<td align="left">baseline_post</td>
<td align="right">15.254758</td>
</tr>
<tr class="even">
<td>sample.kidney_14_baseline_pre</td>
<td align="left">14</td>
<td align="left">baseline_pre</td>
<td align="right">37.207727</td>
</tr>
<tr class="odd">
<td>sample.kidney_15_baseline_post</td>
<td align="left">15</td>
<td align="left">baseline_post</td>
<td align="right">13.139552</td>
</tr>
<tr class="even">
<td>sample.kidney_15_baseline_pre</td>
<td align="left">15</td>
<td align="left">baseline_pre</td>
<td align="right">37.565014</td>
</tr>
<tr class="odd">
<td>sample.kidney_16_baseline_post</td>
<td align="left">16</td>
<td align="left">baseline_post</td>
<td align="right">23.509750</td>
</tr>
<tr class="even">
<td>sample.kidney_16_baseline_pre</td>
<td align="left">16</td>
<td align="left">baseline_pre</td>
<td align="right">38.203048</td>
</tr>
<tr class="odd">
<td>sample.kidney_17_baseline_post</td>
<td align="left">17</td>
<td align="left">baseline_post</td>
<td align="right">4.411050</td>
</tr>
<tr class="even">
<td>sample.kidney_17_baseline_pre</td>
<td align="left">17</td>
<td align="left">baseline_pre</td>
<td align="right">37.731035</td>
</tr>
<tr class="odd">
<td>sample.kidney_18_baseline_post</td>
<td align="left">18</td>
<td align="left">baseline_post</td>
<td align="right">14.734395</td>
</tr>
<tr class="even">
<td>sample.kidney_18_baseline_pre</td>
<td align="left">18</td>
<td align="left">baseline_pre</td>
<td align="right">34.592928</td>
</tr>
<tr class="odd">
<td>sample.kidney_19_baseline_post</td>
<td align="left">19</td>
<td align="left">baseline_post</td>
<td align="right">3.049479</td>
</tr>
<tr class="even">
<td>sample.kidney_19_baseline_pre</td>
<td align="left">19</td>
<td align="left">baseline_pre</td>
<td align="right">37.095411</td>
</tr>
<tr class="odd">
<td>sample.kidney_20_baseline_post</td>
<td align="left">20</td>
<td align="left">baseline_post</td>
<td align="right">16.851456</td>
</tr>
<tr class="even">
<td>sample.kidney_20_baseline_pre</td>
<td align="left">20</td>
<td align="left">baseline_pre</td>
<td align="right">31.596650</td>
</tr>
<tr class="odd">
<td>sample.kidney_21_baseline_post</td>
<td align="left">21</td>
<td align="left">baseline_post</td>
<td align="right">3.494093</td>
</tr>
<tr class="even">
<td>sample.kidney_21_baseline_pre</td>
<td align="left">21</td>
<td align="left">baseline_pre</td>
<td align="right">36.670486</td>
</tr>
<tr class="odd">
<td>sample.kidney_22_baseline_post</td>
<td align="left">22</td>
<td align="left">baseline_post</td>
<td align="right">3.542999</td>
</tr>
<tr class="even">
<td>sample.kidney_22_baseline_pre</td>
<td align="left">22</td>
<td align="left">baseline_pre</td>
<td align="right">28.310753</td>
</tr>
<tr class="odd">
<td>sample.kidney_23_baseline_post</td>
<td align="left">23</td>
<td align="left">baseline_post</td>
<td align="right">3.524408</td>
</tr>
<tr class="even">
<td>sample.kidney_23_baseline_pre</td>
<td align="left">23</td>
<td align="left">baseline_pre</td>
<td align="right">37.322141</td>
</tr>
<tr class="odd">
<td>sample.kidney_24_baseline_post</td>
<td align="left">24</td>
<td align="left">baseline_post</td>
<td align="right">11.445690</td>
</tr>
<tr class="even">
<td>sample.kidney_24_baseline_pre</td>
<td align="left">24</td>
<td align="left">baseline_pre</td>
<td align="right">28.211771</td>
</tr>
<tr class="odd">
<td>sample.kidney_25_baseline_post</td>
<td align="left">25</td>
<td align="left">baseline_post</td>
<td align="right">11.453034</td>
</tr>
<tr class="even">
<td>sample.kidney_25_baseline_pre</td>
<td align="left">25</td>
<td align="left">baseline_pre</td>
<td align="right">24.934383</td>
</tr>
<tr class="odd">
<td>sample.kidney_26_baseline_post</td>
<td align="left">26</td>
<td align="left">baseline_post</td>
<td align="right">28.857246</td>
</tr>
<tr class="even">
<td>sample.kidney_26_baseline_pre</td>
<td align="left">26</td>
<td align="left">baseline_pre</td>
<td align="right">27.860759</td>
</tr>
<tr class="odd">
<td>sample.kidney_27_baseline_post</td>
<td align="left">27</td>
<td align="left">baseline_post</td>
<td align="right">14.729848</td>
</tr>
<tr class="even">
<td>sample.kidney_27_baseline_pre</td>
<td align="left">27</td>
<td align="left">baseline_pre</td>
<td align="right">35.940841</td>
</tr>
<tr class="odd">
<td>sample.kidney_28_baseline_post</td>
<td align="left">28</td>
<td align="left">baseline_post</td>
<td align="right">14.675554</td>
</tr>
<tr class="even">
<td>sample.kidney_28_baseline_pre</td>
<td align="left">28</td>
<td align="left">baseline_pre</td>
<td align="right">28.178635</td>
</tr>
<tr class="odd">
<td>sample.kidney_29_baseline_post</td>
<td align="left">29</td>
<td align="left">baseline_post</td>
<td align="right">28.788652</td>
</tr>
<tr class="even">
<td>sample.kidney_30_baseline_post</td>
<td align="left">30</td>
<td align="left">baseline_post</td>
<td align="right">18.839448</td>
</tr>
<tr class="odd">
<td>sample.kidney_30_baseline_pre</td>
<td align="left">30</td>
<td align="left">baseline_pre</td>
<td align="right">24.448409</td>
</tr>
<tr class="even">
<td>sample.kidney_31_baseline_post</td>
<td align="left">31</td>
<td align="left">baseline_post</td>
<td align="right">4.439941</td>
</tr>
<tr class="odd">
<td>sample.kidney_31_baseline_pre</td>
<td align="left">31</td>
<td align="left">baseline_pre</td>
<td align="right">35.922339</td>
</tr>
<tr class="even">
<td>sample.kidney_32_baseline_post</td>
<td align="left">32</td>
<td align="left">baseline_post</td>
<td align="right">4.611549</td>
</tr>
<tr class="odd">
<td>sample.kidney_32_baseline_pre</td>
<td align="left">32</td>
<td align="left">baseline_pre</td>
<td align="right">30.455091</td>
</tr>
<tr class="even">
<td>sample.kidney_33_baseline_post</td>
<td align="left">33</td>
<td align="left">baseline_post</td>
<td align="right">15.234127</td>
</tr>
<tr class="odd">
<td>sample.kidney_33_baseline_pre</td>
<td align="left">33</td>
<td align="left">baseline_pre</td>
<td align="right">25.032899</td>
</tr>
<tr class="even">
<td>sample.kidney_34_baseline_post</td>
<td align="left">34</td>
<td align="left">baseline_post</td>
<td align="right">4.359224</td>
</tr>
<tr class="odd">
<td>sample.kidney_34_baseline_pre</td>
<td align="left">34</td>
<td align="left">baseline_pre</td>
<td align="right">30.551365</td>
</tr>
<tr class="even">
<td>sample.kidney_35_baseline_post</td>
<td align="left">35</td>
<td align="left">baseline_post</td>
<td align="right">6.274882</td>
</tr>
<tr class="odd">
<td>sample.kidney_35_baseline_pre</td>
<td align="left">35</td>
<td align="left">baseline_pre</td>
<td align="right">30.838598</td>
</tr>
<tr class="even">
<td>sample.kidney_36_baseline_post</td>
<td align="left">36</td>
<td align="left">baseline_post</td>
<td align="right">4.345997</td>
</tr>
<tr class="odd">
<td>sample.kidney_36_baseline_pre</td>
<td align="left">36</td>
<td align="left">baseline_pre</td>
<td align="right">36.929690</td>
</tr>
<tr class="even">
<td>sample.kidney_37_baseline_post</td>
<td align="left">37</td>
<td align="left">baseline_post</td>
<td align="right">20.928901</td>
</tr>
<tr class="odd">
<td>sample.kidney_37_baseline_pre</td>
<td align="left">37</td>
<td align="left">baseline_pre</td>
<td align="right">35.107001</td>
</tr>
<tr class="even">
<td>sample.kidney_38_baseline_post</td>
<td align="left">38</td>
<td align="left">baseline_post</td>
<td align="right">14.790778</td>
</tr>
<tr class="odd">
<td>sample.kidney_38_baseline_pre</td>
<td align="left">38</td>
<td align="left">baseline_pre</td>
<td align="right">33.732354</td>
</tr>
<tr class="even">
<td>sample.kidney_39_baseline_post</td>
<td align="left">39</td>
<td align="left">baseline_post</td>
<td align="right">17.537571</td>
</tr>
<tr class="odd">
<td>sample.kidney_39_baseline_pre</td>
<td align="left">39</td>
<td align="left">baseline_pre</td>
<td align="right">30.476172</td>
</tr>
<tr class="even">
<td>sample.kidney_40_baseline_post</td>
<td align="left">40</td>
<td align="left">baseline_post</td>
<td align="right">3.428423</td>
</tr>
<tr class="odd">
<td>sample.kidney_40_baseline_pre</td>
<td align="left">40</td>
<td align="left">baseline_pre</td>
<td align="right">28.065158</td>
</tr>
<tr class="even">
<td>sample.kidney_41_baseline_post</td>
<td align="left">41</td>
<td align="left">baseline_post</td>
<td align="right">10.472033</td>
</tr>
<tr class="odd">
<td>sample.kidney_41_baseline_pre</td>
<td align="left">41</td>
<td align="left">baseline_pre</td>
<td align="right">27.043803</td>
</tr>
<tr class="even">
<td>sample.kidney_42_baseline_post</td>
<td align="left">42</td>
<td align="left">baseline_post</td>
<td align="right">15.210156</td>
</tr>
<tr class="odd">
<td>sample.kidney_42_baseline_pre</td>
<td align="left">42</td>
<td align="left">baseline_pre</td>
<td align="right">24.953453</td>
</tr>
</tbody>
</table>
</div>
<div id="two-dimensional-case" class="section level2">
<h2><span class="header-section-number">4.3</span> Two-dimensional case</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_cell_trajectory</span>(human, <span class="dt">color_by =</span> <span class="st">&quot;time&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/2-covar-traj-1.png" width="672" /></p>
<p>The sample rank and corresponding pseudo-time:</p>
<pre><code>## 07 21 22 09 17 04 26 10 35 23 34 19 15 31 28 03 29 32 01 02 13 20 40 36 16 18 38 11 05 24 25 14 27 41 30 06 42 12 08 39 33 37</code></pre>
<pre><code>## 0 5.51 6.25 7.56 9.01 9.77 26.39 26.4 30.68 33.46 33.55 33.72 40.93 43.48 45.92 46.7 47.11 47.85 48.75 48.9 48.93 49.58 53.56 53.96 56.18 61.76 64.31 65.68 67.09 72.56 73.62 75.71 86.21 86.42 86.43 86.7 87.21 87.43 87.89 88.13 88.93 89.98</code></pre>
<p>Or reversily:</p>
<pre><code>## 37 33 39 08 12 42 06 30 41 27 14 25 24 05 11 38 18 16 36 40 20 13 02 01 32 29 03 28 31 15 19 34 23 35 10 26 04 17 09 22 21 07</code></pre>
<pre><code>## 0 1.05 1.85 2.09 2.55 2.77 3.28 3.55 3.56 3.77 14.27 16.36 17.42 22.89 24.31 25.67 28.22 33.8 36.02 36.42 40.4 41.05 41.08 41.23 42.13 42.87 43.28 44.06 46.5 49.05 56.26 56.43 56.53 59.3 63.58 63.59 80.21 80.97 82.42 83.73 84.47 89.98</code></pre>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-R-monocle">
<p>Trapnell, Cole. 2017. <em>Monocle: Clustering, Differential Expression, and Trajectory Analysis for Single- Cell Rna-Seq</em>. <a href="https://bioconductor.org/packages/release/bioc/html/monocle.html" class="uri">https://bioconductor.org/packages/release/bioc/html/monocle.html</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quality-control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="homologs-reduction-of-de-gene-set.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
