<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A Report Example</title>
  <meta name="description" content="My first attemp to write scientific report via bookdown.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="A Report Example" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="My first attemp to write scientific report via bookdown." />
  <meta name="github-repo" content="Yaphets-Bo/kidneyproject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Report Example" />
  
  <meta name="twitter:description" content="My first attemp to write scientific report via bookdown." />
  

<meta name="author" content="Bo Sun">


<meta name="date" content="2018-01-02">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="quality-control.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequsites</a></li>
<li class="chapter" data-level="2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html"><i class="fa fa-check"></i><b>2</b> t-SNE analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-human"><i class="fa fa-check"></i><b>2.1</b> Dataset Human</a><ul>
<li class="chapter" data-level="2.1.1" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#post-samples-form-a-distinguishable-cluster"><i class="fa fa-check"></i><b>2.1.1</b> POST samples form a distinguishable cluster</a></li>
<li class="chapter" data-level="2.1.2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#independent-t-sne-on-each-time-point"><i class="fa fa-check"></i><b>2.1.2</b> Independent t-SNE on each time point</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="t-sne-analysis.html"><a href="t-sne-analysis.html#dataset-mouse"><i class="fa fa-check"></i><b>2.2</b> Dataset Mouse</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a><ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#gf"><i class="fa fa-check"></i><b>3.1</b> gene filtering</a></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#check-log-nomality"><i class="fa fa-check"></i><b>3.2</b> check log-nomality</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html"><i class="fa fa-check"></i><b>4</b> Pseudo-time trajectory prediction</a><ul>
<li class="chapter" data-level="4.1" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-1-2-set-ordering-genes-and-reduce-demisionality"><i class="fa fa-check"></i><b>4.1</b> Step 1-2: Set ordering genes and reduce demisionality</a></li>
<li class="chapter" data-level="4.2" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#step-3-ordering-the-cells-in-pseudotime"><i class="fa fa-check"></i><b>4.2</b> Step 3: ordering the cells in pseudotime</a></li>
<li class="chapter" data-level="4.3" data-path="pseudo-time-trajectory-prediction.html"><a href="pseudo-time-trajectory-prediction.html#two-dimensional-case"><i class="fa fa-check"></i><b>4.3</b> Two-dimensional case</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html"><i class="fa fa-check"></i><b>5</b> Homologs Reduction of DE gene set</a><ul>
<li class="chapter" data-level="5.1" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#download-orthology-information-from-biomart"><i class="fa fa-check"></i><b>5.1</b> Download orthology information from BioMart</a></li>
<li class="chapter" data-level="5.2" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#focus-de-genes-of-human-and-mouse-on-homologous-genes-only"><i class="fa fa-check"></i><b>5.2</b> Focus DE genes of Human and Mouse on homologous genes only</a><ul>
<li class="chapter" data-level="5.2.1" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#take-iri2h-iri4h-as-early-response"><i class="fa fa-check"></i><b>5.2.1</b> Take IRI2h + IRI4H as “early response”</a></li>
<li class="chapter" data-level="5.2.2" data-path="homologs-reduction-of-de-gene-set.html"><a href="homologs-reduction-of-de-gene-set.html#take-iri2h-as-early-reponse-only"><i class="fa fa-check"></i><b>5.2.2</b> Take IRI2h as early reponse only</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><i class="fa fa-check"></i><b>6</b> Heatmap Viewing of Informative Genes Used to Predict Sample Trajectory</a><ul>
<li class="chapter" data-level="6.1" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html#selecting-genes"><i class="fa fa-check"></i><b>6.1</b> Selecting genes</a></li>
<li class="chapter" data-level="6.2" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html#considering-the-absolute-distance-between-genes"><i class="fa fa-check"></i><b>6.2</b> Considering the absolute distance between genes</a></li>
<li class="chapter" data-level="6.3" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html#enhanced-heatmap-pay-more-attention-to-shape-of-the-curve."><i class="fa fa-check"></i><b>6.3</b> Enhanced heatmap: pay more attention to “shape” of the “curve”.</a></li>
<li class="chapter" data-level="6.4" data-path="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html"><a href="heatmap-viewing-of-informative-genes-used-to-predict-sample-trajectory.html#examine-certain-genes"><i class="fa fa-check"></i><b>6.4</b> Examine certain genes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Report Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="t-sne-analysis" class="section level1">
<h1><span class="header-section-number">2</span> t-SNE analysis</h1>
<div id="dataset-human" class="section level2">
<h2><span class="header-section-number">2.1</span> Dataset Human</h2>
<p>Dataset Summary: total 163 columns rpkm data<br />
kidney 20, 41, 42 miss 1 year<br />
kidney 29 miss pre<br />
kidney 19 miss 3 months</p>
<div id="post-samples-form-a-distinguishable-cluster" class="section level3">
<h3><span class="header-section-number">2.1.1</span> POST samples form a distinguishable cluster</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tbl &lt;-<span class="st"> </span><span class="kw">tbl_df</span>(tsne<span class="op">$</span>Y)
tbl<span class="op">$</span>persons =<span class="st"> </span>persons
tbl<span class="op">$</span>time_point =<span class="st"> </span>Labels
tbl <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> V1, <span class="dt">y =</span> V2, <span class="dt">label =</span> persons, <span class="dt">col =</span> time_point)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>()</code></pre></div>
<div class="figure"><span id="fig:human"></span>
<img src="_main_files/figure-html/human-1.png" alt="**Similarity between samples and time points(backgroud deducted).**" width="672" />
<p class="caption">
Figure 2.1: <strong>Similarity between samples and time points(backgroud deducted).</strong>
</p>
</div>
<p>In figure <a href="t-sne-analysis.html#fig:human">2.1</a>: <strong>pre</strong> expression profile(<span class="math inline">\(log(FPKM + 1)\)</span>) was deducted from <strong>post</strong>, <strong>1 year</strong>, and <strong>3 months</strong>. And then the dimension reduction was done using <a href="https://lvdmaaten.github.io/tsne/">t-SNE</a> algorithm.<br />
The plot makes more sense now: later time points(<strong>3 months</strong>, <strong>1 year</strong>) stick together and are separated from <strong>post</strong> points, upon which a <em>Pseudotime Sample Trajectory</em> was constructed by means of <a href="http://cole-trapnell-lab.github.io/monocle-release/docs/#constructing-single-cell-trajectories">Monocle 2</a>.</p>
</div>
<div id="independent-t-sne-on-each-time-point" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Independent t-SNE on each time point</h3>
<p>It looks like there is two cluster in <strong>POST</strong> sample.</p>
<pre><code>## Read the 41 x 41 data matrix successfully!
## Using no_dims = 2, perplexity = 6.000000, and theta = 0.500000
## Computing input similarities...
## Normalizing input...
## Building tree...
##  - point 0 of 41
## Done in 0.00 seconds (sparsity = 0.621059)!
## Learning embedding...
## Iteration 50: error is 63.805382 (50 iterations in 0.01 seconds)
## Iteration 100: error is 68.058653 (50 iterations in 0.00 seconds)
## Iteration 150: error is 63.996707 (50 iterations in 0.01 seconds)
## Iteration 200: error is 69.092029 (50 iterations in 0.01 seconds)
## Iteration 250: error is 62.928042 (50 iterations in 0.00 seconds)
## Iteration 300: error is 1.955468 (50 iterations in 0.00 seconds)
## Iteration 350: error is 1.455246 (50 iterations in 0.00 seconds)
## Iteration 400: error is 1.291848 (50 iterations in 0.00 seconds)
## Iteration 450: error is 1.143067 (50 iterations in 0.00 seconds)
## Iteration 500: error is 0.998625 (50 iterations in 0.00 seconds)
## Iteration 550: error is 0.868421 (50 iterations in 0.00 seconds)
## Iteration 600: error is 0.762655 (50 iterations in 0.00 seconds)
## Iteration 650: error is 0.709098 (50 iterations in 0.00 seconds)
## Iteration 700: error is 0.669030 (50 iterations in 0.00 seconds)
## Iteration 750: error is 0.634407 (50 iterations in 0.00 seconds)
## Iteration 800: error is 0.624864 (50 iterations in 0.00 seconds)
## Fitting performed in 0.08 seconds.</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x
## $y, : font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<div class="figure"><span id="fig:human-post-only"></span>
<img src="_main_files/figure-html/human-post-only-1.png" alt="**t-SNE on POST samples only**" width="672" />
<p class="caption">
Figure 2.2: <strong>t-SNE on POST samples only</strong>
</p>
</div>
<pre><code>## Read the 40 x 40 data matrix successfully!
## Using no_dims = 2, perplexity = 6.000000, and theta = 0.500000
## Computing input similarities...
## Normalizing input...
## Building tree...
##  - point 0 of 40
## Done in 0.00 seconds (sparsity = 0.597500)!
## Learning embedding...
## Iteration 50: error is 63.697473 (50 iterations in 0.00 seconds)
## Iteration 100: error is 63.009840 (50 iterations in 0.00 seconds)
## Iteration 150: error is 63.315584 (50 iterations in 0.01 seconds)
## Iteration 200: error is 59.099152 (50 iterations in 0.00 seconds)
## Iteration 250: error is 67.647811 (50 iterations in 0.01 seconds)
## Iteration 300: error is 2.173040 (50 iterations in 0.00 seconds)
## Iteration 350: error is 1.815526 (50 iterations in 0.00 seconds)
## Iteration 400: error is 1.165833 (50 iterations in 0.00 seconds)
## Iteration 450: error is 0.992057 (50 iterations in 0.00 seconds)
## Iteration 500: error is 0.690721 (50 iterations in 0.00 seconds)
## Iteration 550: error is 0.532175 (50 iterations in 0.00 seconds)
## Iteration 600: error is 0.472545 (50 iterations in 0.00 seconds)
## Iteration 650: error is 0.444015 (50 iterations in 0.00 seconds)
## Iteration 700: error is 0.564632 (50 iterations in 0.00 seconds)
## Iteration 750: error is 0.527065 (50 iterations in 0.00 seconds)
## Iteration 800: error is 0.470169 (50 iterations in 0.00 seconds)
## Fitting performed in 0.08 seconds.</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x
## $y, : font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<div class="figure"><span id="fig:human-month-only"></span>
<img src="_main_files/figure-html/human-month-only-1.png" alt="**t-SNE on 3 Month samples only**" width="672" />
<p class="caption">
Figure 2.3: <strong>t-SNE on 3 Month samples only</strong>
</p>
</div>
<pre><code>## Read the 38 x 38 data matrix successfully!
## Using no_dims = 2, perplexity = 5.000000, and theta = 0.500000
## Computing input similarities...
## Normalizing input...
## Building tree...
##  - point 0 of 38
## Done in 0.00 seconds (sparsity = 0.519391)!
## Learning embedding...
## Iteration 50: error is 68.873762 (50 iterations in 0.00 seconds)
## Iteration 100: error is 71.104464 (50 iterations in 0.01 seconds)
## Iteration 150: error is 65.638997 (50 iterations in 0.00 seconds)
## Iteration 200: error is 62.558994 (50 iterations in 0.00 seconds)
## Iteration 250: error is 67.570975 (50 iterations in 0.01 seconds)
## Iteration 300: error is 2.132609 (50 iterations in 0.00 seconds)
## Iteration 350: error is 1.492879 (50 iterations in 0.00 seconds)
## Iteration 400: error is 1.049527 (50 iterations in 0.00 seconds)
## Iteration 450: error is 0.910425 (50 iterations in 0.00 seconds)
## Iteration 500: error is 0.802701 (50 iterations in 0.00 seconds)
## Iteration 550: error is 0.634487 (50 iterations in 0.00 seconds)
## Iteration 600: error is 0.590204 (50 iterations in 0.00 seconds)
## Iteration 650: error is 0.586557 (50 iterations in 0.00 seconds)
## Iteration 700: error is 0.577414 (50 iterations in 0.00 seconds)
## Iteration 750: error is 0.551946 (50 iterations in 0.00 seconds)
## Iteration 800: error is 0.524491 (50 iterations in 0.00 seconds)
## Fitting performed in 0.07 seconds.</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x
## $y, : font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<div class="figure"><span id="fig:human-year-only"></span>
<img src="_main_files/figure-html/human-year-only-1.png" alt="**t-SNE on 1 Year samples only**" width="672" />
<p class="caption">
Figure 2.4: <strong>t-SNE on 1 Year samples only</strong>
</p>
</div>
<pre><code>## Read the 78 x 50 data matrix successfully!
## Using no_dims = 2, perplexity = 15.000000, and theta = 0.500000
## Computing input similarities...
## Normalizing input...
## Building tree...
##  - point 0 of 78
## Done in 0.00 seconds (sparsity = 0.719921)!
## Learning embedding...
## Iteration 50: error is 56.771105 (50 iterations in 0.01 seconds)
## Iteration 100: error is 56.812182 (50 iterations in 0.01 seconds)
## Iteration 150: error is 59.184410 (50 iterations in 0.01 seconds)
## Iteration 200: error is 58.124633 (50 iterations in 0.01 seconds)
## Iteration 250: error is 58.228015 (50 iterations in 0.01 seconds)
## Iteration 300: error is 1.936942 (50 iterations in 0.01 seconds)
## Iteration 350: error is 1.390766 (50 iterations in 0.01 seconds)
## Iteration 400: error is 1.063703 (50 iterations in 0.01 seconds)
## Iteration 450: error is 0.831150 (50 iterations in 0.01 seconds)
## Iteration 500: error is 0.714567 (50 iterations in 0.01 seconds)
## Iteration 550: error is 0.524069 (50 iterations in 0.01 seconds)
## Iteration 600: error is 0.423816 (50 iterations in 0.01 seconds)
## Iteration 650: error is 0.406842 (50 iterations in 0.01 seconds)
## Iteration 700: error is 0.404124 (50 iterations in 0.01 seconds)
## Iteration 750: error is 0.404819 (50 iterations in 0.01 seconds)
## Iteration 800: error is 0.399680 (50 iterations in 0.01 seconds)
## Fitting performed in 0.20 seconds.</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database

## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x
## $y, : font family not found in Windows font database</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## font family not found in Windows font database</code></pre>
<div class="figure"><span id="fig:human-year-month"></span>
<img src="_main_files/figure-html/human-year-month-1.png" alt="**t-SNE on 1 Year and 3 month samples**" width="672" />
<p class="caption">
Figure 2.5: <strong>t-SNE on 1 Year and 3 month samples</strong>
</p>
</div>
<p>But there is no sub-cluster behavior on <strong>3 months</strong> and <strong>1 year</strong> samples, and the <em>two-cluster</em> pattern in <strong>POST</strong> does not persist.</p>
</div>
</div>
<div id="dataset-mouse" class="section level2">
<h2><span class="header-section-number">2.2</span> Dataset Mouse</h2>
<pre><code>## Read the 49 x 49 data matrix successfully!
## Using no_dims = 2, perplexity = 9.000000, and theta = 0.500000
## Computing input similarities...
## Normalizing input...
## Building tree...
##  - point 0 of 49
## Done in 0.00 seconds (sparsity = 0.670554)!
## Learning embedding...
## Iteration 50: error is 56.664756 (50 iterations in 0.01 seconds)
## Iteration 100: error is 51.474400 (50 iterations in 0.01 seconds)
## Iteration 150: error is 56.221535 (50 iterations in 0.01 seconds)
## Iteration 200: error is 54.908899 (50 iterations in 0.01 seconds)
## Iteration 250: error is 56.325644 (50 iterations in 0.01 seconds)
## Iteration 300: error is 1.607589 (50 iterations in 0.00 seconds)
## Iteration 350: error is 0.545783 (50 iterations in 0.00 seconds)
## Iteration 400: error is 0.351512 (50 iterations in 0.01 seconds)
## Iteration 450: error is 0.204200 (50 iterations in 0.01 seconds)
## Iteration 500: error is 0.186298 (50 iterations in 0.00 seconds)
## Fitting performed in 0.06 seconds.</code></pre>
<div class="figure"><span id="fig:mouse"></span>
<img src="_main_files/figure-html/mouse-1.png" alt="Same t-SNE analysis for mouse data." width="672" />
<p class="caption">
Figure 2.6: Same t-SNE analysis for mouse data.
</p>
</div>
<p>As shown in the Figure <a href="t-sne-analysis.html#fig:mouse">2.6</a>:</p>
<ol style="list-style-type: decimal">
<li><p>The syntaxis of same time-point and same mouse-type are stronger in this Mouse data, though it has less sample points here–49 samples in total, divided into three types(IRI, NORM, SHAM), each time point with 3 replicate except for 6mN has 4.</p></li>
<li><p>The perplexity = 9 and set.seed(81) were tuned mannualy by hands-and-eyes. Based on tens of runs, the results are alike–the clustering is good. Perplexity &gt;5 and &lt;= 10 generally works here for limitied times of run done by myself.</p></li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
